servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/LoadBalancerReadySubscriber.java

```diff
@@ -58,31 +57,29 @@ public void onNext(final Object o) {
         if (o instanceof LoadBalancerReadyEvent) {
             LoadBalancerReadyEvent event = (LoadBalancerReadyEvent) o;
             if (event.isReady()) {
-                Processor onHostsAvailable = LoadBalancerReadySubscriber.this.onHostsAvailable;
+                Processor onHostsAvailable = this.onHostsAvailable;
                 if (onHostsAvailable != null) {
-                    LoadBalancerReadySubscriber.this.onHostsAvailable = null;
+                    this.onHostsAvailable = null;
                     onHostsAvailable.onComplete();
                 }
-            } else if (LoadBalancerReadySubscriber.this.onHostsAvailable == null) {
-                LoadBalancerReadySubscriber.this.onHostsAvailable = newCompletableProcessor();
+            } else if (onHostsAvailable == null) {
+                onHostsAvailable = newCompletableProcessor();
             }
         }
     }
 
     @Override
     public void onError(final Throwable t) {
-        Processor onHostsAvailable = LoadBalancerReadySubscriber.this.onHostsAvailable;
+        Processor onHostsAvailable = this.onHostsAvailable;
         if (onHostsAvailable != null) {
-            LoadBalancerReadySubscriber.this.onHostsAvailable = null;
             onHostsAvailable.onError(t);
         }
     }
 
     @Override
     public void onComplete() {
-        Processor onHostsAvailable = LoadBalancerReadySubscriber.this.onHostsAvailable;
+        Processor onHostsAvailable = this.onHostsAvailable;
         if (onHostsAvailable != null) {
-            LoadBalancerReadySubscriber.this.onHostsAvailable = null;
             // Let the load balancer or retry strategy fail any pending requests.
             onHostsAvailable.onComplete();
```

Comment:
Looks like here is a bug. Javadoc for `onHostAvailable()` says:

> Get `Completable` that will complete when a `LoadBalancerReadyEvent` returns `true` from `LoadBalancerReadyEvent#isReady()`.

If `this.onHostsAvailable != null`, LB is not ready. Completing it here will be misleading. We should terminate `onHostsAvailable` with an error to signal that LB events stream terminated.